### PR TITLE
Tilt angles operator

### PR DESCRIPTION
**Pull request by mathturtle**
@cryos This should help with #459 and #445.  This branch makes the 'Mark as Tilt Series', 'Mark as Volume' and 'Set Tilt Angles' operations into Operators in the pipeline.

### CODE REVIEW

**Comment by cryos** [comment]
Great, I will take a look this morning.

**Comment by mathturtle** [comment]
New commit added in response to @Hovden's comment on #459.  An improvement in the responsiveness of the application in many cases, but doesn't have anything else to do with this branch (put here to avoid conflicts later).

File: tomviz/DataSource.cxx [hunk 1]
```diff
@@ -92,6 +92,7 @@ class DataSource::DSInternals
 
 namespace {
 
+/*
```
Comment (cryos) [inline tomviz/DataSource.cxx]: Shouldn't we just delete this, and the one below if it is no longer needed?

**Comment by cryos** [comment]
Looks good to me, I think you should just delete the commented out code if it is no longer needed - it will be in the history if we ever need to resurrect it.

**Comment by mathturtle** [comment]
Updated to remove the code rather than commenting it out.  One thing I'm not sure of... if you load an old state file into this version of tomviz, what will happen.  I tested a little bit and it loads correctly, but I think some of the bugs that motivated this change will still happen in that case.

**Comment by cryos** [comment]
Commented out will not help, but if we need a legacy code path then we should consider it. We need to build up a set of state files to guarantee behavior, and it is something that would be worthwhile after M&M.

**Comment by Hovden** [comment]
Don't worry about legacy state files. Users can package older tomviz versions with datasets for reproducibility.  tomviz is too young to worry about backwards compatibility.

**Pull request merged**